### PR TITLE
Remove cached docker client repo on error

### LINKED ISSUE #3272: Image import controller needs to discard expired cached Docker Hub tokens
It looks like the Hub tokens expire after about 2 hours. The image import controller caches tokens indefinitely. If you run `oc import-image` for the same image stream multiple times, it will stop working after about 2 hours. The only way to reimport again is to restart the master.

### PR DESCRIPTION
**Pull request by ncdc**
If the docker client has a repo cached, it currently caches the auth
token. The token appears to expire after about 2 hours. Remove the repo
from the cache when we get an error so on the next attempt, it will
request a new token.

Fixes #3272

### CODE REVIEW

**Comment by ncdc** [comment]
@smarterclayton PTAL

File: pkg/dockerregistry/client.go [hunk 1]
```diff
@@ -243,6 +243,7 @@ func (c *connection) getRepository(name string) (*repository, error) {
 		}
 		return nil, convertConnectionError(c.url.String(), fmt.Errorf("error getting X-Docker-Token from %s: %v", name, err))
 	}
+	defer resp.Body.Close()
```
Comment (ncdc) [inline pkg/dockerregistry/client.go]: I added this as I assume it's good behavior

**Comment by ncdc** [comment]
[test]

File: pkg/dockerregistry/client.go (hunk 1 above)
Comment (ncdc) [inline pkg/dockerregistry/client.go]: And I added 2 others that were missing

**Comment by ncdc** [comment]
[test]

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**